src/server/services/wms/qgslayerrestorer.cpp

```diff
@@ -21,48 +21,50 @@
 #include "qgsrasterrenderer.h"
 #include "qgsmaplayerstylemanager.h"
 
-QgsLayerRestorer::QgsLayerRestorer( const QList<QgsMapLayer *> &layers )
+QgsLayerRestorer::QgsLayerRestorer( )
 {
-  Q_FOREACH ( QgsMapLayer *layer, layers )
-  {
-    QgsLayerSettings settings;
-    settings.name = layer->name();
 
-    QString style = layer->styleManager()->currentStyle();
-    settings.mNamedStyle = layer->styleManager()->currentStyle();
+}
 
-    // set a custom property allowing to keep in memory if a SLD file has
-    // been loaded for rendering
-    layer->setCustomProperty( "readSLD", false );
+void QgsLayerRestorer::push( QgsMapLayer *layer )
```

Comment:
- ok for `append` instead of `push`
- I don't think the `remove` method brings any value? Being symmetric is nice when the usage is symmetric too. Currently (and for the foreseable future) we only want to add layers so I believe we can skip the `remove` func - but I don't mind adding it if you feel it could be useful :)
- maybe I could add an accessor for `QgsLayerRestorer.::mLayerSettings.keys()` and remove the `QList<QgsMapLayer *> layers;` returned from `QgsRenderer::sldStylizedLayers` and `QgsRenderer::stylizedLayers` because they have the same lifetime.
Their signature would become:

 ```c++
std::unique_ptr<QgsLayerRestorer*> sldStylizedLayers(const QString &sld);
std::unique_ptr<QgsLayerRestorer*> stylizedLayers(const QList<QgsWmsParametersLayer> &params);
 ```

WDYT?

Reply:
> I don't think the remove method brings any value? Being symmetric is nice when the usage is symmetric too. Currently (and for the foreseable future) we only want to add layers so I believe we can skip the remove func - but I don't mind adding it if you feel it could be useful :)

I don't have a strong opinion, I leave you to decide.

> Their signature would become:

 ```c++
std::unique_ptr<QgsLayerRestorer*> sldStylizedLayers(const QString &sld);
std::unique_ptr<QgsLayerRestorer*> stylizedLayers(const QList<QgsWmsParametersLayer> &params);
 ```

:+1: